src/cbc_sdk/platform/admin_grants.py

```diff
@@ -0,0 +1,550 @@
+
+            Returns:
+                ProfileBuilder: This object.
+            """
+            self._roles = roles_list
+            return self
+
+        def add_role(self, role):
+            """
+            Adds a role identifier to the list of roles associated with the new profile.
+
+            Args:
+                role (str): URN of the role to add.
+
+            Returns:
+                ProfileBuilder: This object.
+            """
+            self._roles.append(role)
+            return self
+
+        def set_conditions(self, conditions_structure):
+            """
+            Sets the access conditions associated with the new profile.
+
+            Args:
+                conditions_structure (dict): The conditions associated with the new profile, with 'cidr', 'expiration',
+                                             and 'disabled' members.
+
+            Returns:
+                ProfileBuilder: This object.
+            """
+            self._conditions = conditions_structure
+            return self
+
+        def set_cidr(self, cidr_list):
+            """
+            Sets the list of CIDR access rules associated with the new profile.
+
+            Args:
+                cidr_list (list): List of CIDR access rules for the profile.
+
+            Returns:
+                ProfileBuilder: This object.
+            """
+            self._conditions['cidr'] = cidr_list
+            return self
+
+        def add_cidr(self, cidr):
+            """
+            Adds a CIDR access rule to the new profile.
+
+            Args:
+                cidr (str): CIDR access rule to be added.
+
+            Returns:
+                ProfileBuilder: This object.
+            """
+            self._conditions['cidr'].append(cidr)
+            return self
+
+        def set_expiration(self, expiration):
+            """
+            Sets the expiration time on the new profile.
+
+            Args:
+                expiration (str): The expiration time, specified as ISO 8601.
+
+            Returns:
+                ProfileBuilder: This object.
+            """
+            self._conditions['expiration'] = expiration
+            return self
+
+        def set_disabled(self, flag):
+            """
+            Sets whether or not the new profile is disabled.
+
+            Args:
+                flag (bool): True if this profile is disabled, False if noe.
+
+            Returns:
+                ProfileBuilder: This object.
+            """
+            self._conditions['disabled'] = flag
+            return self
+
+        def set_can_manage(self, flag):
+            """
+            Sets the management flag for the new profile.
+
+            Args:
+                flag (bool): New value of the management flag.
+
+            Returns:
+                ProfileBuilder: This object.
+            """
+            self._can_manage = flag
+            return self
+
+        def build(self):
```

Comment:
This is a different pattern.  The `new_profile()` method on the `Grant` returns a `ProfileBuilder`, which is used to set up all the data that the new profile will contain.  The `build()` method then takes that data and uses it to build an actual `Profile`.  I don't know if we use this pattern elsewhere in the SDK, but it's one that I'm familiar with in other contexts.

Reply:
Given this is a new builder class I am open to using build but we should be cognizant of existing methods in the CBC SDK and reuse existing ones over creating new mechanisms. This seems like a good one to create new though given create is not something we have done much off.

Reply:
Besides which, since you told me to change `new_profile()` to `create_profile()`, that means we would have, effectively, a control flow like `grant.create_profile().{builder operations}.create()`. Using "create" twice would be confusing.